tests/numerics/fv/test_fvutils.py

```diff
@@ -0,0 +1,224 @@
+    * The function for obtaining indices of cells and faces for a partial update of the
+    discretization stencil. 
+
+"""
+from __future__ import division
+
+import pytest
+
+import numpy as np
+import scipy.sparse as sps
+
+from porepy.applications.test_utils.arrays import compare_arrays
+from porepy.grids import simplex, structured
+from porepy.numerics.fv import fvutils
+
+
+@pytest.fixture
+def g_2d():
+    return structured.CartGrid([5, 5])
+
+
+@pytest.fixture
+def g_3d():
+    return structured.CartGrid([3, 3, 3])
+
+
+def test_subcell_topology_2d_cart():
+    # Verify that the subcell topology is correct for a 2d Cartesian grid.
+    x = np.ones(2, dtype=int)
+    g = structured.CartGrid(x)
+    subcell_topology = fvutils.SubcellTopology(g)
+
+    assert np.all(subcell_topology.cno == 0)
+
+    ncum = np.bincount(subcell_topology.nno, weights=np.ones(subcell_topology.nno.size))
+    assert np.all(ncum == 2)
+
+    fcum = np.bincount(subcell_topology.fno, weights=np.ones(subcell_topology.fno.size))
+    assert np.all(fcum == 2)
+
+    # There is only one cell, thus only unique subfno
+    usubfno = np.unique(subcell_topology.subfno)
+    assert usubfno.size == subcell_topology.subfno.size
+
+    assert np.all(np.in1d(subcell_topology.subfno, subcell_topology.subhfno))
+
+
+def test_subcell_mapping_2d_simplex():
+    # Verify that the subcell mapping is correct for a 2d simplex grid.
+    p = np.array([[0, 1, 1, 0], [0, 0, 1, 1]])
+    g = simplex.TriangleGrid(p)
+
+    subcell_topology = fvutils.SubcellTopology(g)
+
+    ccum = np.bincount(subcell_topology.cno, weights=np.ones(subcell_topology.cno.size))
+    assert np.all(ccum == 6)
+
+    ncum = np.bincount(subcell_topology.nno, weights=np.ones(subcell_topology.nno.size))
+    assert ncum[0] == 2
+    assert ncum[1] == 4
+    assert ncum[2] == 2
+    assert ncum[3] == 4
+
+    fcum = np.bincount(subcell_topology.fno, weights=np.ones(subcell_topology.fno.size))
+    assert np.sum(fcum == 4) == 1
+    assert np.sum(fcum == 2) == 4
+
+    subfcum = np.bincount(
+        subcell_topology.subfno, weights=np.ones(subcell_topology.subfno.size)
+    )
+    assert np.sum(subfcum == 2) == 2
+    assert np.sum(subfcum == 1) == 8
+
+
+@pytest.mark.parametrize(
+    "grid, expected_eta",
+    [(simplex.StructuredTriangleGrid([1, 1]), 1 / 3), (structured.CartGrid([1, 1]), 0)],
+)
+def test_determine_eta(grid, expected_eta):
+    # Test that the automatic computation of the pressure continuity point is
+    # correct for a Cartesian and simplex grid.
+    assert fvutils.determine_eta(grid) == expected_eta
+
+
+def test_diagonal_scaling_matrix():
+    # Generate a matrix with a known row sum, check that the target function returns the
+    # correct diagonal.
+    A = np.array([[1, 2, 3], [0, -5, 6], [-7, 8, 0]])
+    A_sum = np.array([6, 11, 15])
+    values = 1 / A_sum
+
+    D = fvutils.diagonal_scaling_matrix(sps.csr_matrix(A))
+    assert compare_arrays(values, D.diagonal())
+
```

Comment:
```suggestion
"""IMPLEMENTATION NOTE: Below are tests that considers the definition of computational stencils under
a partial update of the FV discretization. The logic of the tests is:
1) Construct a grid, and pick a cell quantity (node, face, cell) as the trigger for refinement
2) Compute the update stencil
3) Compare the stencil with a known value, hard-coded based on knowledge of the grid and the ordering of the cell quantities.
```
I added a short explanation, though you will need to reformat the comment to make it fit in the line etc.